### PR TITLE
release-notes.txt: add 2020.07 release notes

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This are the release notes for the upcoming 2020.07 release. Please read and ensure that your of the past 3 months work is addressed.

### Testing procedure
Read.

### CODE REVIEW

File: release-notes.txt [hunk 1]
```diff
@@ -1,3 +1,775 @@
...
+* samd5x: allow to use XOSC as clock source
+* saml1x: pm: set deep flag
++ stm32:
+  + add QDEC support for CPU_FAM_STM32F1
+  + add non-blocking UART
+  + implement reset to bootloader
+  * filter availability of RNG by CPU not by board
+  * stm32_common: make RTT_FREQUENCY configurable
+  + stm32f4: Add DMA config for SPI
+  + stm32g4: add support
```
Comment (bergzand) [inline release-notes.txt:216]: I'm missing something along the lines of
```
  + Decrease latency in SPI DMA path
```

Something that describes the effect of both https://github.com/RIOT-OS/RIOT/pull/14096 and https://github.com/RIOT-OS/RIOT/pull/14097 in a single note. Feel free to reword my suggestion to something that makes sense for the reader

File: release-notes.txt [hunk 2]
```diff
@@ -1,3 +1,775 @@
...
+RIOT is developed by an international open-source community which is
+independent of specific vendors (e.g. similarly to the Linux community) and is
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release:
+===================
+
+The 2020.07 release includes put a focus on the improvement of automated
```
Comment (aabadie) [inline release-notes.txt:21]:
```suggestion
The 2020.07 release puts a focus on the improvement of automated
```
?

File: release-notes.txt [hunk 3]
```diff
@@ -1,3 +1,775 @@
...
+testing. In that vain various new CI integrations such as CircleCI for
+documentation building and online presentation and Github Actions to check the
+tooling of RIOT where merged. The `riotctrl_shell` python module allows to
+abstract shell commands as python methods for testing using the newly created
+`riotctrl` python package [1]. Kconfig migration reached phase 2 with various
+board features being exposed to Kconfig. New network protocols were ported for
+RIOT such as MQTT—in form of the `paho-mqtt` package and the lookup client
+component for CoRE RD `cord_lc`. The OpenWSN network stack with 6TSCH support
```
Comment (aabadie) [inline release-notes.txt:29]:
```suggestion
component for CoRE RD `cord_lc`. The OpenWSN network stack with 6TiSCH support
```

File: release-notes.txt [hunk 4]
```diff
@@ -1,3 +1,775 @@
...
+was reintegrated into RIOT. Suppotr for several new boards and new sensors was
```
Comment (aabadie) [inline release-notes.txt:30]:
```suggestion
was reintegrated into RIOT. Support for several new boards and new sensors was
```

File: release-notes.txt (hunk 1 above)
Comment (miri64) [reply release-notes.txt:217]: See my latest change.

File: release-notes.txt [hunk 5]
```diff
@@ -1,3 +1,775 @@
...
+  + add black magic probe config
+  + add missing vendor files
+  + implement EasyDMA-based SPI peripheral implemenation
++ nrf52811: initial support
+* nrf5x_common:
+  * add configurable RTT_FREQUENCY
+  * enable bias correction in hwrng
+* sam0_common:
+  + add DMA peripheral driver
+  + add support for saml21, samd5x
```
Comment (benpicco) [inline release-notes.txt:200]:
```suggestion
  + add PWM support for saml21, samd5x
```

File: release-notes.txt [hunk 6]
```diff
@@ -1,3 +1,775 @@
...
+  + implement periph/dac
+  * Make SPI peripheral DMA compatible
+  * move PWM to common code
```
Comment (benpicco) [inline release-notes.txt:203]:
```suggestion
```
the location of `pwm.c` isn't that noteworthy

File: release-notes.txt [hunk 7]
```diff
@@ -1,3 +1,775 @@
...
+    pn532, rn2xx3, slipdev, sps30, tcs37727, tmp00x
++ networking: Expose configurations to Kconfig
+  + asymcute, coap, csma
+  + gnrc:
+    + gomach, lwmac, mac, pktdump, pktbuf, rpl, tcp
++ usbus/cdc/ecm: Expose configurations to Kconfig
+
+API Changes
+===========
+- cpu/sam0_common: drop prescaler from timer config
```
Comment (benpicco) [inline release-notes.txt:293]:
```suggestion
- cpu/sam0_common: drop prescaler from timer config
- cpu/sam0_common: PWM config now requires GCLK source 
```

Comment (miri64) [reply release-notes.txt:293]: Which PR was that?

Comment (miri64) [reply release-notes.txt:293]: Also https://github.com/RIOT-OS/RIOT/pull/14007? Then the label for API change was missing ;-)

File: release-notes.txt [hunk 8]
```diff
@@ -1,3 +1,777 @@
...
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release:
+===================
+
+The 2020.07 release puts a focus on the improvement of automated testing. In
+that vain various new CI integrations such as CircleCI for documentation
+building and online presentation and Github Actions to check the tooling of
+RIOT where merged. The `riotctrl_shell` python module allows to abstract shell
```
Comment (leandrolanzieri) [inline release-notes.txt:24]:
```suggestion
RIOT were merged. The `riotctrl_shell` python module allows to abstract shell
```

File: release-notes.txt [hunk 9]
```diff
@@ -1,3 +1,777 @@
...
++ sdp3x : Add driver for sdp3x sensor
+
+Build System / Tooling
+----------------------
++ make:
+  + add code generator targets
+  + add JLink support for stm32
+  * docker.inc.mk: add and use generic docker_run_make
+  + allow multiple external board dirs
+  + info-global.inc.mk: add info-boards-features-blacklisted
```
Comment (leandrolanzieri) [inline release-notes.txt:249]: Do you think we should include #14351?

Comment (miri64) [reply release-notes.txt:249]: Done

**Comment by miri64** [comment]
Squashed the current state.

File: release-notes.txt (hunk 7 above)
Comment (miri64) [reply release-notes.txt:294]: Clarified offline. It was indeed part of #14007

File: release-notes.txt [hunk 10]
```diff
@@ -1,3 +1,780 @@
...
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release:
+===================
+
+The 2020.07 release puts a focus on the improvement of automated testing. In
+that vain various new CI integrations such as CircleCI for documentation
+building and online presentation and Github Actions to check the tooling of
+RIOT were merged. The `riotctrl_shell` python module allows to abstract shell
```
Comment (aabadie) [inline release-notes.txt:24]:
```suggestion
RIOT when merged. The `riotctrl_shell` python module allows to abstract shell
```

Comment (miri64) [reply release-notes.txt]: Addressed and squashed directly.

File: release-notes.txt [hunk 11]
```diff
@@ -1,3 +1,780 @@
...
+independent of specific vendors (e.g. similarly to the Linux community) and is
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release:
+===================
+
+The 2020.07 release puts a focus on the improvement of automated testing. In
+that vain various new CI integrations such as CircleCI for documentation
```
Comment (chrysn) [inline release-notes.txt:22]:
```suggestion
that vein, various new CI integrations such as CircleCI for documentation
```

File: release-notes.txt [hunk 12]
```diff
@@ -1,3 +1,780 @@
...
+building and online presentation and Github Actions to check the tooling of
```
Comment (chrysn) [inline release-notes.txt:23]:
```suggestion
building and online presentation, and Github Actions to check the tooling of
```
(eases the reading flow given there's already more strongly binding "and")

File: release-notes.txt [hunk 13]
```diff
@@ -1,3 +1,780 @@
...
+RIOT when merged. The `riotctrl_shell` python module allows to abstract shell
+commands as python methods for testing using the newly created `riotctrl`
+python package [1]. Kconfig migration reached phase 2 with various board
+features being exposed to Kconfig. New network protocols were ported for RIOT
+such as MQTT—in form of the `paho-mqtt` package and the lookup client
```
Comment (chrysn) [inline release-notes.txt:28]:
```suggestion
such as MQTT (in form of the `paho-mqtt` package) and the lookup client
```

File: release-notes.txt [hunk 14]
```diff
@@ -1,3 +1,780 @@
...
+
+API Changes
+===========
+- cpu/sam0_common: drop prescaler from timer config
+- cpu/sam0_common: PWM config now requires GCLK source
+- drivers/mtd: Change API to return 0 on success
+- drivers/mtd_mapper: count offset in sectors
+- net/sock_dtls: provide getter function for UDP sock
+- net/sock_dtls: add timeout to sock_dtls_send and add sock_dtls_session_init
+
```
Comment (chrysn) [inline release-notes.txt:300]: Does this large diff belong here? It's way more detailed than all the rest, and looks a bit like a copy-pasting slip.

Comment (miri64) [reply release-notes.txt:300]: The rest are either simple changes or additions. This change is a larger API overhaul and thus justifies a more detailed listing of the changes.

**Comment by miri64** [comment]
Addressed and squashed @chrysn's comments.

File: release-notes.txt (hunk 14 above)
Comment (miri64) [reply release-notes.txt:300]: (see also larger API changes in other releases below)

**Comment by miri64** [comment]
Adapted and amended for RC3. I don't expect an RC4, so last chance for comments until tomorrow.

**Comment by miri64** [comment]
Fixed [error in static tests](https://travis-ci.org/github/RIOT-OS/RIOT/builds/711155635#L153-L154)

File: release-notes.txt (hunk 10 above)
Comment (aabadie) [reply release-notes.txt]: Now that I'm reading this again, the initial word `were` was ok.

**Comment by miri64** [comment]
Backport provided in #14604

**Pull request merged**